sdk/tests/iot/common/test_az_iot_common.c

```diff
@@ -259,6 +258,20 @@ static void test_az_iot_provisioning_client_logging_succeed()
   az_log_set_classifications(NULL);
 }
 
+static void test_az_log_incomplete_list_fails_gracefully()
+{
+  az_log_classification const classifications[] = { AZ_LOG_HTTP_REQUEST };
+  az_log_set_classifications(classifications);
```

Comment:
I don't see in this PR, how we would detect the end of the list consistently, without going to read the rest of the memory until we see a value that coincides with AZ_LOG_END_OF_LIST.

Reply:
I see, so you're checking the classifications in the list, and not the classification bein logged. But in that case we are verifying first half the list (on average) on every log attempt. And still it doesn't prevent you from the AV - the code will still shoot in the memory, given the values are happen to be within the range of defined classifications.